src/Symfony/Component/Yaml/Tests/ParserTest.php

```diff
@@ -1078,6 +1090,58 @@ public function testFloatKeys()
     }
 
     /**
+     * @group legacy
+     * @expectedDeprecation Implicit casting of incompatible mapping keys to strings is deprecated since version 3.3 and will throw \Symfony\Component\Yaml\Exception\ParseException in 4.0. Pass the PARSE_KEYS_AS_STRING flag to explicitly enable the type casts.
+     */
+    public function testBooleanKeys()
+    {
+        $yaml = <<<'EOF'
+true: foo
+false: bar
+EOF;
+
+        $expected = array(
+            1 => 'foo',
+            0 => 'bar',
```

Comment:
That's actually an inconsistency in the current behaviour of the parser for inlined and block mapping (the keys would be parsed as the strings `true` and `false` inside an inlined mapping).